### ISSUE TITLE
How to test the performance of Rhino?

### ISSUE THREAD

**Issue by lmz123321** [microsoft/causica#59]
Hi there.

I wonder which way to test the Rhino model is more reasonable?

Namely:

- Should I use the same full time causal graph to generate all time series in the training set?
- Should I use an extra test set, or should I directly test the performance (e.g. F1 score between the recovered graph and the ground-truth graph) on the training set?
- If an extra test set is required, should I use the same or different causal graph to generate data for the training and test set?

**Comment by WenboGong** [comment]
Thanks for your interests in Rhino. Since Rhino is a causal discovery method, if you are solely interested in the graph discovery performance, there is no "test set". So to test Rhino graph recovery performance, here are the steps:

1. Generate a training timeseries dataset with a ground truth graph (should be the same graph for all training timeseries).
2. Train Rhino on that training dataset.
3. Compare the inferred graph from Rhino with the ground truth graph. 

Hope this answers your question.
